develop to master

andyBzg opened the pull request.

andyBzg committed:
fix:

andyBzg committed:
feat: add schedule task for recurring payments

andyBzg committed:
class rename

andyBzg committed:
initial commit

andyBzg committed:
feat: add method to find managers with a certain status, sort by the number of clients in ascending order

andyBzg committed:
feat: add field and DB-column for ProductType

andyBzg committed:
feat: rename field and column

andyBzg committed:
feat: add method to find a client by uuid and account type

andyBzg committed:
feat: add methods to find a clients Current and Savings accounts

andyBzg committed:
feat: add methods to find a clients Savings agreement

andyBzg committed:
feat: add method and query to find agreement by client uuid and certain ProductType

andyBzg committed:
rename sum to amount

andyBzg committed:
feat: add on new client creation: find the first manager with the least number of clients and assign it to the client
feat: add method to find clients which have current and savings accounts

andyBzg committed:
feat: add method to find clients which have current and savings accounts

andyBzg committed:
feat: add method to find clients which have certain Account Types

andyBzg committed:
feat: add method to find managers sorted by client count in ascending order

andyBzg committed:
feat: new method: work in progress

andyBzg committed:
feat: add scheduling annotation

andyBzg committed:
image updated

luxorious commented on src/main/java/org/crazymages/bankingspringproject/repository/AccountRepository.java:
Optional<Account> findAccountByClientUuidAndType(UUID uuid, AccountType type);

ты каждый раз, когда делаешь запрос в базу данных, вытаскиваешь из оптионала все сущности? если несколько раз нужно найти аккаунт – то много раз нужно прописывать if (optional.isPresent()){} или я что то путаю...

andyBzg replied:
С помощью этого метода я получаю Optional, который содержит сущность или ничего. А в сервисе я избавился от if (optional.isPresent()), заменил это на 
```
return accountRepository.findAccountByClientUuidAndType(uuid, AccountType.SAVINGS)
                .orElseThrow(() -> new DataNotFoundException(String.valueOf(uuid)));
```

luxorious commented on src/main/java/org/crazymages/bankingspringproject/repository/TransactionRepository.java:
как писал мне Даниил - удали все закоментирование строки

andyBzg replied:
Да, этот метод я сегодня пытался заставить работать, но программа постоянно падает, поэтому я временно его закомментировал, чтобы потом попробовать исправить

luxorious commented on src/main/java/org/crazymages/bankingspringproject/service/database/impl/ClientDatabaseServiceImpl.java:
общался с гуру, то она порекомендовала мне сделать отдельно проверку списка на нулл, если он нулл, то выбросить исключение, если он пуст или заполнен - то вернуть список...

andyBzg replied:
Спасибо, записал себе)

andyBzg committed:
initial commit

andyBzg committed:
refactor: package and stereotype

andyBzg committed:
feat: add methods to find present and deleted entities

andyBzg committed:
feat: add new enumeration

andyBzg committed:
initial commit

andyBzg committed:
refactor: minor refactor

andyBzg committed:
feat: add create() method overload, method takes uuid as a parameter

andyBzg committed:
feat: add method to get account statement

andyBzg committed:
feat: add method to get account statement
feat: add ListValidator dependency

andyBzg committed:
feat: add method that finds existing entities
feat: add method that finds entities flagged as deleted
feat: add ListValidator dependency

andyBzg committed:
feat: add method that finds existing entities
feat: add method that finds entities flagged as deleted

andyBzg committed:
feat: add method overload
feat: add method that finds existing entities
feat: add method that finds entities flagged as deleted

andyBzg committed:
feat: add method that finds existing entities
feat: add method that finds entities flagged as deleted
feat: add method that finds product by type, status and currency

andyBzg committed:
feat: add new logic to create()
feat: add method that finds existing entities
feat: add method that finds entities flagged as deleted
feat: add method that finds product by type, status and currency
feat: add ListValidator dependency

andyBzg committed:
feat: add new logic to create()
feat: add new method composeAgreement()
feat: add method that finds existing entities
feat: add method that finds entities flagged as deleted
feat: add method that finds product by type, status and currency
feat: add ListValidator dependency
feat: add EnumTypeMatcher dependency

andyBzg committed:
feat: add method that finds existing entities
feat: add method that finds entities flagged as deleted
feat: add ListValidator dependency
feat: add EnumTypeMatcher dependency
feat: add method that finds managers sorted by product quantity
feat: add method to get first manager from list

andyBzg committed:
feat: add method that finds existing entities
feat: add method that finds entities flagged as deleted
refactor: name of method
feat: add method that finds managers sorted by product quantity
feat: add method to get first manager from list

andyBzg committed:
feat: add method that finds product by type, status and currency code

andyBzg committed:
commit

andyBzg committed:
add javadoc

andyBzg committed:
Merge pull request #4 from andyBzg/week-04-doc

add javadoc

andyBzg committed:
Merge pull request #3 from andyBzg/week-03

Week 03 code review

andyBzg committed:
feat: add caching

andyBzg committed:
Merge pull request #5 from andyBzg/week-04-caching

feat: add caching

andyBzg committed:
refactor

andyBzg committed:
moved ad renamed

andyBzg committed:
initial commit

andyBzg committed:
feat: extract transaction creation to another class

andyBzg committed:
feat: extract agreement creation to another class
feat: refactor for using dto

andyBzg committed:
feat: refactor for using dto

andyBzg committed:
test: add tests

andyBzg committed:
Merge pull request #8 from andyBzg/develop-week-04-dto

dto to week-04

andyBzg committed:
initial commit

andyBzg committed:
initial commit

andyBzg committed:
commit

andyBzg committed:
feat: refactor for better readability

andyBzg committed:
feat: refactor for using dto

andyBzg committed:
Merge pull request #9

dto to week-04

andyBzg committed:
Merge branch 'develop-week-04-dto' into develop-week-04

andyBzg committed:
refactor

andyBzg committed:
Merge remote-tracking branch 'origin/develop-week-04' into develop-week-04

andyBzg committed:
Merge pull request #7 from andyBzg/develop-week-04

week-04 to develop

andyBzg committed:
initial commit

andyBzg committed:
add @PropertySource and property paths

andyBzg committed:
add feature to block client by its UUID

andyBzg committed:
add feature to find accounts where product type and status match.

andyBzg committed:
Add test cases for Service methods

andyBzg committed:
rename and refactor

andyBzg committed:
rename status

andyBzg committed:
add messages for logging that are taken from the property file

andyBzg committed:
rename method call to match the return data type

andyBzg committed:
rename method call to match the return data type

andyBzg committed:
add method to get account with specified product type and status

andyBzg committed:
add methods to get list of agreements and dtos by client uuid

andyBzg committed:
add method to get list of clients by status
add method that blocks client by uuid

andyBzg committed:
refactor

andyBzg committed:
refactor

andyBzg committed:
update documentation for methods

andyBzg committed:
initial commit

andyBzg committed:
add spring webflux dependency

andyBzg committed:
refactor code for better readability

andyBzg committed:
refactor for more versatility

andyBzg committed:
refactor for more versatility

andyBzg committed:
refactor transaction creation
bugfix BigDecimal rounding mode
add documentation

andyBzg committed:
Merge pull request #10 from andyBzg/week-05-scheduler

Scheduler to week-05

andyBzg committed:
Merge pull request #13 from andyBzg/week-05-currency_exchange

currency to scheduler

andyBzg committed:
initial commit

andyBzg committed:
add cron expression for currency rates updater

andyBzg committed:
initial commit

andyBzg committed:
refactor

andyBzg committed:
Merge pull request #12 from andyBzg/week-05-currency_exchange

currency exchanch to week-05

andyBzg committed:
add documentation

andyBzg committed:
add ability to execute transactions between different currencies

andyBzg committed:
fix typo in method name

andyBzg committed:
initial commit

andyBzg committed:
add negative test cases

andyBzg committed:
initial commit

andyBzg committed:
initial commit

andyBzg committed:
add null argument check and refactor

andyBzg committed:
initial commit

andyBzg committed:
refactor for proper usage of DTOs

andyBzg committed:
Merge pull request #16 from andyBzg/tests-dto_refactor

refactor for proper usage of DTOs

andyBzg committed:
initial commit

andyBzg committed:
initial commit

andyBzg committed:
add new test cases

andyBzg committed:
bugfix and refactor variable,
extract method to separate component

andyBzg committed:
remove unnecessary semicolon

andyBzg committed:
Merge pull request #17 from andyBzg/week-05-tests

Week 05 tests ok

andyBzg committed:
add new test cases

andyBzg committed:
bugfix calling the wrong getter

andyBzg committed:
initial commit

andyBzg committed:
initial commit

andyBzg committed:
method refactor to return a value for better testing

andyBzg committed:
initial commit

andyBzg committed:
initial commit

andyBzg committed:
refactor to better readability

andyBzg committed:
Merge pull request #18

week-05-tests

andyBzg committed:
Merge pull request #19

tests-dto_refactor

andyBzg committed:
Merge pull request #14 from andyBzg/week-05-currency

Сurrency to week-05

andyBzg committed:
refactor to TransactionInitializer

andyBzg committed:
refactor to TransactionInitializer

andyBzg committed:
refactor to TransactionInitializer

andyBzg committed:
refactor to AgreementInitializer

andyBzg committed:
rename property file

andyBzg committed:
Add tests for AccountService class

andyBzg committed:
refactor ClientDto.builder() instead of new ClientDto

andyBzg committed:
refactor: controller methods to accept a string as a parameter instead of an uuid

andyBzg committed:
refactor for using builder

andyBzg committed:
refactor for using builder

andyBzg committed:
refactor

andyBzg committed:
add null argument check,refactor for using builder

andyBzg committed:
add new test cases

andyBzg committed:
add argument check and documentation

andyBzg committed:
add argument check and documentation

andyBzg committed:
add argument check

andyBzg committed:
add argument check

andyBzg committed:
add new test cases

andyBzg committed:
add argument check to avoid NullPointerException

andyBzg committed:
add new test cases

andyBzg committed:
refactor arguments check

andyBzg committed:
refactor test method name

andyBzg committed:
add new test cases

andyBzg committed:
add new test cases

andyBzg committed:
refactor test methods

andyBzg committed:
refactor

andyBzg committed:
refactor

andyBzg committed:
add new test cases

andyBzg committed:
add new test cases

andyBzg committed:
add DateTimeParseException handler

andyBzg committed:
add DateTimeParseException message

andyBzg committed:
add new test case

andyBzg committed:
refactor for better readability , add argumetn check

andyBzg committed:
refactor for better readability , add argument check

andyBzg committed:
refactor

andyBzg committed:
refactor test cases

andyBzg committed:
refactor test cases

andyBzg committed:
add web security configuration

andyBzg committed:
add login and logout using swagger

andyBzg committed:
add logout endpoint for using with swagger ui

andyBzg committed:
refactor endpoints, add logging

andyBzg committed:
refactor class name

andyBzg committed:
add springdoc swagger dependencies

andyBzg committed:
refactor queries

andyBzg committed:
add dto mappers for specific cases

andyBzg committed:
add methods for working with basic entities, add documentation

andyBzg committed:
refactor

andyBzg committed:
add null-argument check

andyBzg committed:
refactor

andyBzg committed:
add new test cases

andyBzg committed:
Merge pull request #20 from andyBzg/week-06_dto-builders

Week 06 dto builders

andyBzg committed:
Merge pull request #11 from andyBzg/develop-week-05

week-05 to develop

andyBzg committed:
refactor test cases

andyBzg committed:
add method to retrieve currency exchange rate from DB by currency code

andyBzg committed:
refactor, add method to retrieve currency exchange rate from DB by currency code

andyBzg committed:
add primary key for table, refactor currency_code column to accept only unique value

andyBzg committed:
refactor, add documentation

andyBzg committed:
add new test cases

andyBzg committed:
refactor create method update save currency only if not exists in DB, add logging

andyBzg committed:
refactor properties order

andyBzg committed:
add new schematic representation of database tables

andyBzg committed:
Merge pull request #21

week_06-exchange_rates_refactor

andyBzg committed:
refactor order of packages

andyBzg committed:
add new test cases

andyBzg committed:
refactor mappers

andyBzg committed:
add null argument check

andyBzg committed:
remove class which duplicates other class's logic

andyBzg committed:
refactor test cases

andyBzg committed:
add new test cases

andyBzg committed:
refactor methods, add method to get list of dto

andyBzg committed:
Merge pull request #23

dto_refactor to week_06

andyBzg committed:
add new entities to maintain username, password, roles in database

andyBzg committed:
add new configuration class which sets the default admin

andyBzg committed:
add support of BCryptPasswordEncoder and JdbcUserDetailsManager instead of InMemoryUserDetailsManager

andyBzg committed:
add email column a unique key

andyBzg committed:
add new custom exception

andyBzg committed:
add registration controller, service and dto

andyBzg committed:
add registration service implementation

andyBzg committed:
add ability to load managers credentials

andyBzg committed:
refactor

andyBzg committed:
refactor to use a ManagerDatabaseService instead of a ManagerRepository

andyBzg committed:
modify method to return entities instead of dto

andyBzg committed:
edit test case

andyBzg committed:
add new method for using with entities instead of dtos

andyBzg committed:
refactor method for using with ClientDatabaseService instead of ClientRepository

andyBzg committed:
refactor dependency inversion

andyBzg committed:
refactor for configure only default admin

andyBzg committed:
refactor

andyBzg committed:
refactor

andyBzg committed:
refactor

andyBzg committed:
add endpoint for manager registration

andyBzg committed:
add manager registration service and dto

andyBzg committed:
add method that saves manager entity into database

andyBzg committed:
Merge pull request #25

Merging manager registration into security

andyBzg committed:
Merge pull request #24

updated security will be merged into week-06

andyBzg committed:
Merge pull request #22

week_06 to Develop
